community/kernel/src/test/java/org/neo4j/kernel/impl/index/schema/TemporalIndexCacheTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index.schema;
+
+import org.junit.Test;
+
+import org.neo4j.helpers.collection.Iterables;
+
+import static org.junit.Assert.assertEquals;
+
+public class TemporalIndexCacheTest
+{
+    @Test
+    public void shouldIterateOverCreatedParts() throws Exception
+    {
+        // GIVEN
+        TemporalIndexCache<String, Exception> cache = new TemporalIndexCache<>( new StringFactory() );
+        cache.dateTime();
```

Comment:
This test isn't exhaustive. The point of this commit was to ensure that they were all part of the iteration, right?

Reply:
I mainly wanted to see if the previous implementation returned anything after the first null (it didn't). Now that we have the test however, I agree it could be better. We improved it.